### ISSUE TITLE
Should the share method be directly on Navigator?

### ISSUE THREAD

**Issue by plinss** [w3c/web-share#180]
From the [TAG review](https://github.com/w3ctag/design-reviews/issues/554):

Although this is only a single method, it may be better to add an interface to Navigator containing the share method rather than adding a method directly on Navigator. In the event that other methods or properties are added in the future it would prevent polluting the Navigator namespace and make feature detection easier and more consistent.

**Comment by saschanaz** [comment]
Something like this looks feasible:

```webidl
[Exposed=Window, SecureContext]
interface NavigatorShare {
  Promise<undefined> request(optional ShareData data = {});
  boolean supportsBlob(DOMString mimeType); // based on the proposal from Marcos
};

partial interface Navigator {
  [SecureContext] NavigatorShare webshare; // needs a new name
};

**Comment by marcoscaceres** [comment]
Although arguably right about the namespace pollution and making feature detection easier and more consistent, I'd be reluctant to make drastic changes to the API's location at this point. The API surface is small enough that this is not a huge issue.

**Comment by plinss** [comment]
But is it guaranteed to never get bigger? I already see proposals for more related methods...

**Comment by mgiuca** [comment]
I agree with @marcoscaceres . The API has been shipped in multiple browsers for several years, and is in widespread use. It is _probably_ a better idea to put the share method into its own namespace, but it is not worth making a breaking change to the API at this late stage. Indeed, there might be more methods added to Navigator, but it's a small amount of pollution.

Contrary to this, when we designed the [Badging API](https://github.com/w3c/badging), we went in the opposite direction to Web Share, and proposed a `Badge` namespace. [At TPAC 2019](https://github.com/w3c/badging/issues/55), the working group made the decision to put the four methods of `Badge` directly into the `Navigator` namespace. I don't really know why that is (and I wasn't there), but it is a precedent.

I'm going to close this because it is not practical to make this change.
